src/UX/Identification/Core.php

```diff
@@ -0,0 +1,32 @@
+<?php namespace ILIAS\UX\Identification;
+
+/**
+ * Class DynamicProvider
+ *
+ * @author Fabian Schmid <fs@studer-raimann.ch>
+ */
+class Core implements ProviderInterface {
+
+	/**
+	 * @var string
+	 */
+	protected $class_name = '';
+
+
+	/**
+	 * DynamicProvider constructor.
+	 *
+	 * @param string $class_name
+	 */
+	public function __construct(string $class_name) { $this->class_name = $class_name; }
+
+
+	/**
+	 * @param string $identifier
+	 *
+	 * @return Identification$
+	 */
+	public function internal(string $identifier): IdentificationInterface {
```

Comment:
Why internal? Explain here in one sentence. Would not identification be the better name, since you always get an identification back?